### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by mberacochea**

### CODE REVIEW

File: ena_portal_api/ena_handler.py [hunk 1]
```diff
@@ -200,9 +200,6 @@ def post_request(self, data):
                 self.url, data=data, auth=self.auth, **get_default_connection_headers()
             )
         else:
-            logging.warning(
-                "Not authenticated, set env vars ENA_API_USER and ENA_API_PASSWORD to access private data."  # noqa: E501
-            )
             response = requests.post(
```
Comment (MGS-sails) [inline ena_portal_api/ena_handler.py:203]: Perhaps we could refactor this as such, to have less conditionals:
 
 ```
 def post_request(self, data):
     headers = get_default_connection_headers()
     auth = {'auth': self.auth} if self.auth else {}
     response = requests.post(self.url, data=data, **auth, **headers)
     
     return response
 ```

File: ena_portal_api/ena_handler.py [hunk 2]
```diff
@@ -234,15 +231,13 @@ def get_study(
         data["fields"] = fields or STUDY_DEFAULT_FIELDS
 
         if primary_accession and not secondary_accession:
-            data["query"] = 'study_accession="{}"'.format(primary_accession)
```
Comment (MGS-sails) [inline ena_portal_api/ena_handler.py:237]: I think we could also simplify this as such:

```
def get_study(self, primary_accession=None, secondary_accession=None, fields=None, attempt=0):
    """Retrieve study information based on primary or secondary accession."""
    data = {
        **get_default_params(),
        "result": "read_study",
        "fields": fields or STUDY_DEFAULT_FIELDS,
    }

    query_parts = []
    if primary_accession:
        query_parts.append(f'study_accession="{primary_accession}"')
    if secondary_accession:
        query_parts.append(f'secondary_study_accession="{secondary_accession}"')

    data["query"] = " AND ".join(query_parts)

 
```

**Review by MGS-sails — COMMENTED** [review]
Most of my observations are just around having so many conditionals. 
I think it makes the code hard to follow, and potentially difficult to extend. 
I've left specific comments in a couple of places. 
But I think areas like lines 404-436 and also line 242-257, 
could potentially be improved. Some of the logic should also perhaps be extracted into separate functions. 
But I appreciate that alot of this is probably beyond the scope of this pull request. 
Just something to keep in mind I think

**Comment by mberacochea** [comment]
> Most of my observations are just around having so many conditionals. I think it makes the code hard to follow, and potentially difficult to extend. I've left specific comments in a couple of places. But I think areas like lines 404-436 and also line 242-257, could potentially be improved. Some of the logic should also perhaps be extracted into separate functions. But I appreciate that alot of this is probably beyond the scope of this pull request. Just something to keep in mind I think

Indeed, this source code needs to be refactored (quite a lot). We had a meeting last year, a code review session to work on this code base to improve it. We will include refactoring this one as a track in one of our mini-hackathons.

File: ena_portal_api/ena_handler.py (hunk 1 above)
Comment (mberacochea) [reply ena_portal_api/ena_handler.py:203]: Yes, we will incoporate this later. Thanks!

File: ena_portal_api/ena_handler.py (hunk 2 above)
Comment (mberacochea) [reply ena_portal_api/ena_handler.py:237]: Yep
